### PR TITLE
Stop assigning availability  set unconditionally

### LINKED ISSUE #7357: Create Availability Sets based on users requests in KKP
**User Story**
The KKP api should have a new field called assignAvailabilitySets of a type bool for azure clusters.

**Acceptance Criteria**
- Azure user clusters should have a new cloud field called assignAvailabilitySets. 
- KKP should specify the value of assignAvailabilitySets field in the created machine deployment for azure clusters.

### PR DESCRIPTION
**Pull request by moadqassem**
**What this PR does / why we need it**:
We create availability sets for azure cloud provider unconditionally. On cluster creation we create those av sets and attach them to the clusters vm. This PR creates av set based on the cluster configs. When the value `AssignAvailabilitySet` is set, the av set will be created and vms are assigned to it.
 
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #7357 

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Create and assign azure availability sets on demand
```

### CODE REVIEW

File: pkg/test/e2e/utils/apiclient/models/time.go [hunk 1]
```diff
@@ -5,98 +5,13 @@ package models
...
-// MarshalJSON, and MarshalText methods store the Time.Location's offset, but not
-// the location name. They therefore lose information about Daylight Saving Time.
-//
-// In addition to the required “wall clock” reading, a Time may contain an optional
-// reading of the current process's monotonic clock, to provide additional precision
-// for comparison or subtraction.
-// See the “Monotonic Clocks” section in the package documentation for details.
+// Time Time is a wrapper around time.Time which supports correct
+// marshaling to YAML and JSON.  Wrappers are provided for many
+// of the factory methods that the time package offers.
```
Comment (xrstf) [inline pkg/test/e2e/utils/apiclient/models/time.go]: Why was this file changed in this PR? You didn't touch anything related, so the code generation should not have touched this file, or am I wrong?

Comment (moadqassem) [reply pkg/test/e2e/utils/apiclient/models/time.go]: I don't know tbh

Comment (xrstf) [reply pkg/test/e2e/utils/apiclient/models/time.go]: Could you then simply remove the unrelated changes from this PR and we retest the verify job until it works?

File: pkg/test/e2e/utils/apiclient/models/etcd_restore_status.go [hunk 2]
```diff
@@ -20,7 +20,6 @@ type EtcdRestoreStatus struct {
 	Phase EtcdRestorePhase `json:"phase,omitempty"`
 
 	// restore time
-	// Format: date-time
```
Comment (xrstf) [inline pkg/test/e2e/utils/apiclient/models/etcd_restore_status.go]: Please also remove changes to this file.

File: pkg/test/e2e/utils/apiclient/models/etcd_backup_config_condition.go [hunk 3]
```diff
@@ -25,11 +25,9 @@ type EtcdBackupConfigCondition struct {
 	Reason string `json:"reason,omitempty"`
 
 	// last heartbeat time
-	// Format: date-time
```
Comment (xrstf) [inline pkg/test/e2e/utils/apiclient/models/etcd_backup_config_condition.go]: Please also remove changes to this file.

File: pkg/test/e2e/utils/apiclient/models/backup_status.go [hunk 4]
```diff
@@ -32,86 +32,45 @@ type BackupStatus struct {
 	JobName string `json:"jobName,omitempty"`
 
 	// backup finished time
-	// Format: date-time
```
Comment (xrstf) [inline pkg/test/e2e/utils/apiclient/models/backup_status.go]: Please also remove changes to this file.

**Review by xrstf — APPROVED** [review]
/approve
